src/BinSkim.Rules/RulesExtensionMethods.cs

```diff
@@ -108,6 +132,35 @@
             return sb.ToString();
         }
 
+        public static (string nameOnly, string comment) ExtractNameAndComment(string nameWithComment, bool includeCommentMarkers = false)
```

Comment:
>ExtractNameAndComment [](http://example.com/codeflow?start=56&length=21)

This complexity looks non-obvious to me.

i.e., i'm not sure this addition to the design puts us in a cleaner/more capable condition but rather appears to make some relatively hard to understand code a little harder to understand.